packages/app/src/cli/services/dev/ui/components/TabPanel.tsx

```diff
@@ -0,0 +1,151 @@
+import React, {useState} from 'react'
+import {Box, Text, useInput, useStdin} from '@shopify/cli-kit/node/ink'
+
+export interface Tab {
+  label: string
+  content?: React.ReactNode
+  shortcuts?: TabShortcut[]
+  action?: () => Promise<void>
+}
+
+interface TabShortcut {
+  key: string
+  condition?: () => boolean
+  action: () => Promise<void>
+}
+
+interface TabDisplay extends Tab {
+  inputKey: string
+  header: string
+}
+
+interface TabPanelProps {
+  tabs: {[key: string]: Tab}
+  initialActiveTab?: string
+}
+
+export const TabPanel: React.FunctionComponent<TabPanelProps> = ({tabs, initialActiveTab}) => {
+  const {isRawModeSupported: canUseShortcuts} = useStdin()
+  const firstTabKey = Object.keys(tabs)[0]
+  const [activeTab, setActiveTab] = useState<string | undefined>(initialActiveTab ?? firstTabKey)
```

Comment:
Can this really be undefined? I feel like `TabPanelProps` could have `initialActiveTab` as a mandatory property.